core/src/main/java/com/rudderstack/android/sdk/core/RudderClient.java

```diff
@@ -91,7 +91,7 @@ public static RudderClient getInstance(@NonNull Context context, @Nullable Strin
             RudderLogger.logVerbose("getInstance: instance null. creating instance");
             // assert writeKey is not null or empty
             if (TextUtils.isEmpty(writeKey)) {
-                RudderLogger.logError("RudderClient: getInstance: writeKey can not be null or empty");
+                throw new IllegalArgumentException("RudderClient: getInstance: writeKey can not be null or empty");
```

Comment:
@1abhishekpandey are we doing the same for iOS? If not, implement there as well. But first check what Segment is doing.